src/generate/peripheral.rs

```diff
@@ -16,6 +18,31 @@ use anyhow::{anyhow, bail, Context, Result};
 
 use crate::generate::register;
 
+/// An enum defining how to represent registers in the rendered outputs, binding a bool
+/// which describes whether or not renderers should include type information, 
+/// which allows for disjoint arrays to share the same type.
+#[derive(Debug)]
+enum Repr {
+    ArrayAndInfo(bool),
+    ExpandAndInfo(bool),
+    // String to store the common type when the erc is part of a disjoint array
+    SingleAndInfo(bool, String),
+}
+
+impl Default for Repr {
+    fn default() -> Self { Repr::ExpandAndInfo(false) }
+}
+
+impl fmt::Display for Repr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Repr::ArrayAndInfo(info) => write!(f,"ArrayAndInfo({})",info),
+            Repr::ExpandAndInfo(info) => write!(f,"ExpandAndInfo({})",info),
+            Repr::SingleAndInfo(info,s) => write!(f,"SingleAndInfo({},\"{}\")",info,s),
+        }
+    }
+}
```

Comment:
this looks like it's just a reimplementation of the `Debug` impl

```suggestion
impl fmt::Display for Repr {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{self:?}")
    }
}
```